fleet/comms.go

```diff
@@ -88,6 +88,7 @@ func (svc fleetCommsService) NotifyGroupNewDataset(ctx context.Context, ag Agent
 		Version:   p.Version,
 		Data:      pdata,
 		DatasetID: datasetID,
+		GroupID:   ag.ID,
```

Comment:
This is good but we need to add GroupID to the other places AgentPolicyRPCPayload is created too: NotifyAgentAllDatasets, NotifyGroupPolicyUpdate, NotifyGroupPolicyRemoval and in NotifyAgentAllDatasets there could be multiple GroupIDs